src/System.Runtime/tests/System/Runtime/CompilerServices/ConditionalWeakTableTests.netcoreapp1.1.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+
+using Xunit;
+
+namespace System.Runtime.CompilerServices.Tests
+{
+    public partial class ConditionalWeakTableTests
+    {
+        [Fact]
+        public static void AddOrUpdateDataTest()
+        {
+            ConditionalWeakTable<string, string> cwt = new ConditionalWeakTable<string, string>();
+            string key = "key1";
+            cwt.AddOrUpdate(key, "value1");
+
+            string value;
+            Assert.True(cwt.TryGetValue(key, out value));
+            Assert.Equal(value, "value1");
+            Assert.Equal(value, cwt.GetOrCreateValue(key));
+            Assert.Equal(value, cwt.GetValue(key, k => "value1"));
```

Comment:
This assert isn't necessarily testing what you want it to test.  GetValue will look up key and return its value if it exists; if it doesn't exist, it'll call the specified lambda to create the value for the key, add it to the table, and then return that added value.  But the value returned from the lambda is the same as the one previously stored and expected for the key; if there was a bug that caused GetValue to invoke the lambda even if the key was in the table, this assert would still pass.  The lambda should really return some other value.